src/Model/Notification.php

```diff
@@ -3,721 +3,159 @@
 namespace Friendica\Model;
 
 use Exception;
-use Friendica\App;
+use Friendica\BaseModel;
 use Friendica\Content\Text\BBCode;
 use Friendica\Content\Text\HTML;
-use Friendica\Core\PConfig\IPConfig;
-use Friendica\Core\L10n;
-use Friendica\Core\Protocol;
-use Friendica\Core\System;
 use Friendica\Database\Database;
-use Friendica\DI;
-use Friendica\Protocol\Activity;
+use Friendica\Network\HTTPException\InternalServerErrorException;
 use Friendica\Util\DateTimeFormat;
-use Friendica\Util\Proxy as ProxyUtils;
 use Friendica\Util\Temporal;
-use Friendica\Util\XML;
-use ImagickException;
 use Psr\Log\LoggerInterface;
-use Friendica\Network\HTTPException;
 
 /**
- * Methods for read and write notifications from/to database
- *  or for formatting notifications
+ * Model for an entry in the notify table
+ * - Including additional, calculated properties
+ *
+ * Is used either for frontend interactions or for API-based interaction
+ * @see https://github.com/friendica/friendica/blob/develop/doc/API-Entities.md#notification
+ *
+ * @property string  hash
+ * @property integer type
+ * @property string  name   Full name of the contact subject
+ * @property string  url    Profile page URL of the contact subject
+ * @property string  photo  Profile photo URL of the contact subject
+ * @property string  date   YYYY-MM-DD hh:mm:ss local server time
+ * @property string  msg
+ * @property integer uid  	Owner User Id
+ * @property string  link   Notification URL
+ * @property integer iid  	Item Id
+ * @property integer parent Parent Item Id
+ * @property boolean seen   Whether the notification was read or not.
+ * @property string  verb   Verb URL (@see http://activitystrea.ms)
+ * @property string  otype  Subject type (`item`, `intro` or `mail`)
+ *
+ * @property-read string name_cache Full name of the contact subject
+ * @property-read string msg_cache  Plaintext version of the notification text with a placeholder (`{0}`) for the subject contact's name.
+ *
+ * @property-read integer timestamp  Unix timestamp
+ * @property-read string  dateRel  	 Time since the note was posted, eg "1 hour ago"
+ * @property-read string  $msg_html
+ * @property-read string  $msg_plain
  */
-final class Notification
```

Comment:
Why remove the `final` keyword?

Reply:
Because I noticed that `final` bricks phpunit mocking.. It's such a hard setting that no mocking library is capable of creating a "double" of it... So we should just use the `final` keyword in cases we won't need to mock it ..